lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java

```diff
@@ -695,6 +697,7 @@ else if (deletePrefixResult == DeletePrefixResult.DELETE_KEYS_FAILURE) {
             }
             deleteObject(key + DIRECTORY_SUFFIX);
         }
+        // TODO should we return true also when deleteObject() returned false? (currently deleteObject's return value is never used)
```

Comment:
This behavior has been there for over 8 years now  - see e0fe1c78
Maybe we should use the outcome of delete actually instead of simply returning `true` and offer to the users (temporarily for a few releases) a fallback setting to still return `true` independently of the returned result from S3 on deleting an object.

Reply:
i don't want to decide this now in this PR. It's definitely difficult to change fragile things like this.